spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/GcpDatastoreAutoConfiguration.java

```diff
@@ -102,20 +106,31 @@
 
 	@Bean
 	@ConditionalOnMissingBean
-	public Datastore datastore() {
-		DatastoreOptions.Builder builder = DatastoreOptions.newBuilder()
-				.setProjectId(this.projectId)
-				.setHeaderProvider(new UserAgentHeaderProvider(this.getClass()))
-				.setCredentials(this.credentials);
-		if (this.namespace != null) {
-			builder.setNamespace(this.namespace);
-		}
+	public DatastoreNamespaceProvider namespaceProvider() {
```

Comment:
Do we even need this default bean?

Reply:
I don't think I'm following. Could you explain what you would put instead?

Reply:
Nothing, just deal with the situation when `DatastoreNamespaceProvider` is not present.